libs/synchronization/include/hpx/synchronization/spinlock.hpp

```diff
@@ -71,8 +71,7 @@ namespace hpx { namespace lcos { namespace local {
 
             for (std::size_t k = 0; !acquire_lock(); ++k)
             {
-                util::detail::yield_k(k, "hpx::lcos::local::spinlock::lock",
-                    hpx::threads::pending_boost);
+                util::detail::yield_k(k, "hpx::lcos::local::spinlock::lock");
```

Comment:
What motivated this change?

Reply:
Cleanup. The thread state parameter is there on master for the free functions `yield_k` and `yield_while`, but not passed on to the `agent_base::yield_k` member function. This is for example what `execution_agent` does: https://github.com/STEllAR-GROUP/hpx/blob/d639fba90dfb76ca9711b68fba6284368dccc5c7/src/runtime/threads/execution_agent.cpp#L56-L75. I'll happily go the other way and add back the the parameter to `agent_base::yield_k` (doesn't introduce any problems with dependencies).

Reply:
Good catch! The important piece is however, that when spinning on an HPX spinlock we yield the HPX thread with `priority_boost`. This way the thread will not sit indefinitely long in the queues.

Reply:
Indeed, `priority_boost` is the important part. I think what's in `execution_agent` seems like a fair solution. No need to think about which status to give, but still does something reasonable by default. The thresholds for when to yield in what way may of course need to be tweaked.